en/lessons/typeclasses/string-representation.md

```diff
@@ -1,4 +1,673 @@
+## What Show is and isn't good for
+
+The `Show` class, as we will discuss further below, is specifically designed to display Haskell expressions. It exists primarily to be used by Haskell development tools such as GHCi.
+
+Other types of software that aren't Haskell development tools also often need to produce strings, but typically not in a Haskell-specific format, and so there often exists some means of constructing strings that is more appropriate than `Show` for a given purpose.
+
+The `show` function is commonly used to produce `String` representations of numbers.
+
+```haskell
+{-# language OverloadedStrings, TypeApplications #-}
+
+printDiskUsage :: Float -> IO ()
+printDiskUsage x = putStrLn $
+    "Disk usage is currently " <> show @Float (x * 100) <> "%"
+```
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> printDiskUsage (2 / 3)
+Disk usage is currently 66.66667%
+ghci> printDiskUsage 0.0000000001
+Disk usage is currently 1.0e-8%
+```
+
+We might instead prefer to use [`Text`](https://hackage.haskell.org/package/text) to build and print our messages. In addition to boasting improved efficiency over `String`, the `text` package offers a nice set functions for for displaying [integral](https://hackage.haskell.org/package/text/docs/Data-Text-Lazy-Builder-Int.html) and [floating-point](https://hackage.haskell.org/package/text/docs/Data-Text-Lazy-Builder-RealFloat.html) numbers. Below we give a revised example, this time specifying that we don't want to use scientific notation, and that we only care to see the fractional portion carried out to two decimal places.
+
+```haskell
+{-# language OverloadedStrings, TypeApplications #-}
+
+import qualified Data.Text.Lazy.Builder as Text
+import qualified Data.Text.Lazy.IO as Text
+import Data.Text.Lazy.Builder.RealFloat
+
+printDiskUsage :: Float -> IO ()
+printDiskUsage x = Text.putStrLn $ Text.toLazyText $
+    "Disk usage is currently " <>
+    formatRealFloat @Float
+        Fixed      -- Don't use scientific notation
+        (Just 2)   -- Limit to at most 2 decimal places
+        (x * 100)
+    <>  "%"
+```
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> printDiskUsage (2 / 3)
+Disk usage is currently 66.67%
+ghci> printDiskUsage 0.0000000001
+Disk usage is currently 0.00%
+```
+
+### Exercise: Print a Handle
+
+We discussed `Handle` above. If you are going to write a log message that involves a `Handle`, you probably want to use [`hShow`](https://hackage.haskell.org/package/base/docs/System-IO.html#v:hShow) instead. This function includes more information.
+
+```haskell
+hShow :: Handle -> IO String
+```
+
+Write a program that opens a file handle and uses `hShow` to print information about it to the terminal.
+
+## Deriving Show
+
+### Stock deriving
+
+`Show` can be stock-derived for any enumeration type. We did this without mentioning it in the previous lesson.
+
+```haskell
+data Brightness = Light | Dark
+    deriving Show
+
+data Hue = Red | Green | Blue
+    deriving Show
```

Comment:
It's nice that you expose the reader to "stock derivation". Could you perhaps include the `stock` deriving keyword as well? This is valid for the other instances of `deriving Show`

Reply:
Cool, done.